app/components/exercises/ExerciseList.js

```diff
@@ -13,18 +13,50 @@ import { Table, TableHeader, TableHeaderColumn,
 import { StepsRaisedButton, StepsActionButton } from '../material-style'
 import { textDark, errorText } from '../colors';
 
+import ConfirmDialog from '../widgets/ConfirmDialog';
+
 
 // -=-=-=-=-=-= COMPONENT =-=-=-=-=-=-
 
-const ExerciseList = ({ user, exercises, deleteExercise }) => (
+
+export class ExerciseList extends Component {
+
+  constructor(props) {
+    super(props)
+    this.state = {
+      confirmOpen: false,
+      exerciseId: "",
+      userId: ""
+    }
+    this.dialogClose = this.dialogClose.bind(this);
+    this.dialogOpen = this.dialogOpen.bind(this);
+  }
+
+  dialogClose(){
+    this.setState({confirmOpen: false});
```

Comment:
This and below function should contain a single `setState` call, this is needlessly overloading the state reonciler